### PR TITLE
https://github.com/qooxdoo/qooxdoo/issues/10133: move mangle-privates to compile.json

### PR DESCRIPTION
**Pull request by hkollmann**

### CODE REVIEW

File: source/resource/qx/tool/schema/compile-1-0-0.json [hunk 1]
```diff
@@ -247,29 +268,34 @@
             "type": ["boolean","string"],
             "description": "When true, output the typescript definition into ``./compiled/source/qooxdoo.d.ts`. If you use a string instead of true, the string is the name of the file which is generated inside the target output directory"
           },
+          "mangle-privates": {
+            "description": " Whether and how to mangle private identifiers. Can be overridden on a per application basis. Can be overridden on a per application basis.",
```
Comment (derrell) [inline source/resource/qx/tool/schema/compile-1-0-0.json:272]: Repeated sentence "Can be overridden on a per application basis"

Also, it should be "per-application" (with a dash) here and other places where that phrase is used

**Review by derrell — CHANGES_REQUESTED** [review]

**Review by cboulanger — APPROVED** [review]

**Review by johnspackman — APPROVED** [review]
Approving this change (thank you @hkollmann ) , my only thought is that perhaps in future command line options which change the output should be removed altogether and only allowed in compile.js?

Years ago I think the idea was that you could compile just with command line and no compile.js at all, but that's obviously not viable (not any more anyway), and I question the benefit of leaving them in.

**Comment by hkollmann** [comment]
@johnspackman : You can compile without compile.js! Most of my projects do so.

**Comment by johnspackman** [comment]
> @johnspackman : You can compile without compile.js! Most of my projects do so.

Oh wow!  OK I take it all back! :) 👍

**Comment by oetiker** [comment]
according to semantic versioning, the version of the schema should change when it ismodified .. since this change is compatible, the version number would be 1-1-0 ...

**Comment by hkollmann** [comment]
@oetiker : this changes are introduced here https://github.com/qooxdoo/qooxdoo/commit/1b6c45e5b2a4658030e52608c1b26e9f6c50670d#diff-2c50702c8b657fb45f0b13b2b56b8f8ec10ad4a87a76ed43d2b7292ef4152795. But where lost.

We discussed the version change sometime ago the result that it will bring us into an update hell...

**Comment by oetiker** [comment]
> @oetiker : this changes are introduced here [1b6c45e#diff-2c50702c8b657fb45f0b13b2b56b8f8ec10ad4a87a76ed43d2b7292ef4152795](https://github.com/qooxdoo/qooxdoo/commit/1b6c45e5b2a4658030e52608c1b26e9f6c50670d#diff-2c50702c8b657fb45f0b13b2b56b8f8ec10ad4a87a76ed43d2b7292ef4152795). But where lost.
> 
> We discussed the version change sometime ago the result that it will bring us into an update hell...

oh ... missed that one ... well can't help you there ... I can't do more than provide all the tools for sustainability if the mode is to ignore it ... well ..

**Review by oetiker — APPROVED** [review]

**Comment by johnspackman** [comment]
>> We discussed the version change sometime ago the result that it will bring us into an update hell...

>if the mode is to ignore it ... well ..

It's a temporary issue - the compiler requires the current version of compile.json and Manifest.json, so if we change the version number updates then everything needs to be changed (eg Manifest.json means that every plugin will need to have the schema version changed).  For users who switch between compiler versions, this means that the plugin versions have to be switched back and forth also, for no good reason. 

One day there would be a simple mapping between version numbers, eg a v1.0.0 can be translated into v2.0.0 on the fly so that the compiler can work

**Review by derrell — APPROVED** [review]

**Review by level420 — APPROVED** [review]

**Review by cboulanger — APPROVED** [review]

**Pull request merged**